src/main/java/ch/ethz/seb/sebserver/webservice/servicelayer/lms/impl/olat/OlatLmsAPITemplate.java

```diff
@@ -272,56 +323,101 @@ private LmsSetupTestResult testLmsSetupSettings() {
+        return new SEBRestriction(Long.valueOf(id), r.configKeys, r.browserExamKeys, new HashMap<String, String>());
+    }
 
-        return Result.tryCatch(() -> {
+    private SEBRestriction setRestrictionForAssignmentId(final RestTemplate restTemplate, String id, SEBRestriction restriction) {
+        final String url = String.format("/restapi/assessment_modes/%s/seb_restriction", id);
+        final RestrictionDataPost post = new RestrictionDataPost();
+        post.browserExamKeys = new ArrayList<>(restriction.browserExamKeys);
+        post.configKeys = new ArrayList<>(restriction.configKeys);
+        final RestrictionData r = this.apiPost(restTemplate, url, post, RestrictionDataPost.class, RestrictionData.class);
+        return new SEBRestriction(Long.valueOf(id), r.configKeys, r.browserExamKeys, new HashMap<String, String>());
+    }
 
-            // TODO get the SEB client restrictions that are currently set on the remote LMS for
-            //      the given quiz / course derived from the given exam
+    private SEBRestriction deleteRestrictionForAssignmentId(final RestTemplate restTemplate, String id) {
+        final String url = String.format("/restapi/assessment_modes/%s/seb_restriction", id);
+        final RestrictionData r = this.apiDelete(restTemplate, url, RestrictionData.class);
+        // OLAT returns RestrictionData with null values upon deletion.
+        // We return it here for consistency, even though SEB server does not need it
+        return new SEBRestriction(Long.valueOf(id), r.configKeys, r.browserExamKeys, new HashMap<String, String>());
+    }
 
-            throw new RuntimeException("TODO");
-        });
+    @Override
+    public Result<SEBRestriction> getSEBClientRestriction(final Exam exam) {
+        return Result.of(getRestTemplate()
+                .map(t -> this.getRestrictionForAssignmentId(t, exam.externalId))
+                .getOrThrow());
     }
 
     @Override
     public Result<SEBRestriction> applySEBClientRestriction(
             final String externalExamId,
             final SEBRestriction sebRestrictionData) {
-
-        return Result.tryCatch(() -> {
-
-            // TODO apply the given sebRestrictionData settings as current SEB client restriction setting
-            //      to the remote LMS for the given quiz / course.
-            //      Mainly SEBRestriction.configKeys and SEBRestriction.browserExamKeys
-
-            throw new RuntimeException("TODO");
-        });
+        return Result.of(getRestTemplate()
+                .map(t -> this.setRestrictionForAssignmentId(t, externalExamId, sebRestrictionData))
+                .getOrThrow());
     }
 
     @Override
     public Result<Exam> releaseSEBClientRestriction(final Exam exam) {
         @SuppressWarnings("unused")
         final String quizId = exam.externalId;
 
-        return Result.tryCatch(() -> {
+        return Result.of(getRestTemplate()

```

Comment:
Here you should be able to use something like this instead:

`return getRestTemplate()
                .map(t -> this.deleteRestrictionForAssignmentId(t, exam.externalId))
                .map(x -> exam);`

You don't have to create a new Result if the return type of the first call is a Result itself can just map the following calls. This also ensures that exceptions that are thrown during the calls are catched correctly and delivered with the Results error field when happened.